### ISSUE TITLE
kpasswd silently fails if passwords mismatch

### ISSUE THREAD

**Issue by lhoward** [heimdal/heimdal#469]
`UI_UTIL_read_pw_string()` failing to match the new passwords results in nothing being printed to the user. This is confusing. Most `passwd` programs will print an error.

```
% kpasswd
foobar@LUKKTONE.COM's Password: 
New password: 
Verify password - New password: 
% 
```

**Comment by lhoward** [comment]
Exit code is 1 though

**Comment by lhoward** [comment]
`UI_UTIL_read_pw_string()` may need a distinct error code for password mismatch

**Comment by lhoward** [comment]
Looking at the OpenSSL code that `UI_UTIL_read_pw_string()` purports to emulate, the verify failure message should be reported by the function itself rather than the caller.

**Comment by lhoward** [comment]
merged

**Comment by lhoward** [comment]
merged in 65ed504d

**Comment by lhoward** [comment]
This fix now causes kadmin to double report password mismatches

```
kadmin> cpw lukeh
lukeh@LUKKTONE.COM's Password: 
Verify password - lukeh@LUKKTONE.COM's Password: 
Verify failure
kadmin: cpw lukeh: Password mismatch
```

**Comment by nicowilliams** [comment]
OpenSSL did add the third argument, and they call it `verify`.  I don't mind the extension.

**Comment by lhoward** [comment]
Fixed again in 014f1688
